### ISSUE TITLE
Can we get a new binary release please?

### ISSUE THREAD

**Issue by shayne-fletcher** [commercialhaskell/stack#5516]
ghc-lib CI is currently flying blind with respect to ghc-9.0.1 due to 2.5.1 needing https://github.com/commercialhaskell/stack/pull/5508 which appears now to be on HEAD.

I can't even take advantage of the fix locally because of https://github.com/commercialhaskell/stack/issues/5491.

The ideal resolution would be a binary release.

**Comment by essandess** [comment]
The lack of a new binary release is affecting numerous downstream issues. A partial list:
* https://github.com/macports/macports-ports/pull/10086
* https://github.com/macports/macports-ports/pull/10514
* https://trac.macports.org/ticket/61422
* https://trac.macports.org/ticket/62579

**Comment by domenkozar** [comment]
cc @qrilka I'd also appreciate a release for with the Nix fixes.

**Comment by borsboom** [comment]
I've published a release candidate: https://github.com/commercialhaskell/stack/releases/tag/rc%2Fv2.7.0.1.  Please try it out and let us know if you have any trouble.

@domenkozar This was just built from the `master` branch, so it should include the nix fixes that were merged a week or so back.

**Comment by essandess** [comment]
> I've published a release candidate: https://github.com/commercialhaskell/stack/releases/tag/rc%2Fv2.7.0.1. Please try it out and let us know if you have any trouble.
> 
> @domenkozar This was just built from the `master` branch, so it should include the nix fixes that were merged a week or so back.

Thank you!

I'm unable to bootstrap om macOS 11. Hitting these issues, log attached:
```
:info:build hpack                 > /opt/local/var/macports/build/_opt_local_ports_lang_stack/stack/work/.tmp/stack-fea8a3c87f5826c7/hpack-0.33.0/<built-in>:15:10: error:
:info:build hpack                 >      warning: non-portable path to file '".stack-work/dist/x86_64-osx/Cabal-2.4.0.1/build/Hpack/autogen/cabal_macros.h"'; specified path differs in case from file name on disk [-Wnonportable-include-path]
…
:info:build [ 41 of 111] Compiling Stack.Constants
:info:build <command line>: can't load framework: Cocoa (not found)
:info:build Completed 161 action(s).
:info:build --  While building package stack-2.7.0.1 (scroll up to its section to see the error) using:
:info:build       /opt/local/var/macports/build/_opt_local_ports_lang_stack/stack/work/stack-rc-v2.7.0.1/.stack-work/dist/x86_64-osx/Cabal-2.4.0.1/setup/setup --builddir=.stack-work/dist/x86_64-osx/Cabal-2.4.0.1 build lib:stack exe:stack --ghc-options ""
:info:build     Process exited with code: ExitFailure 1
```

Log file: [main.log](https://github.com/commercialhaskell/stack/files/6371105/main.log)

**Comment by essandess** [comment]
This appears to be the same issue as #5491.

**Comment by essandess** [comment]
> I've published a release candidate: https://github.com/commercialhaskell/stack/releases/tag/rc%2Fv2.7.0.1. Please try it out and let us know if you have any trouble.

On a positive note, I verify that the binary release is able to build `pandoc` on macOS 11.

**Comment by asr** [comment]
I could also compile `Agda` (master repository) using the binary release candidate.

**Comment by essandess** [comment]
MacPorts merged this rc because it addresses important issues with the old release, and is now able to build its entire Haskell stack stack over a broad range of macOS architectures: http://packages.macports.org/

**Comment by borsboom** [comment]
I've just made another release candidate: https://github.com/commercialhaskell/stack/releases/tag/rc%2Fv2.7.0.3.  The only change from the previous release candidate is that it's built with GHC 8.10.4 and so should be fix the issues with building it on macOS Big Sur.

**Comment by essandess** [comment]
Thanks! This works. PR on MacPorts for this bootstrappable version: https://github.com/macports/macports-ports/pull/10859

Mostly out of curiosity, but also perhaps because it may [?] be good to be able to build `stack` without bootstrapping, would you please post the steps to building `stack` from `ghc`, not `stack` itself.

**Comment by borsboom** [comment]
It's not something we have official instructions for, but the basic idea is that you'd first have to install cabal-install (it has a bootstrap script), and then use cabal-install to build a bootstrap `stack` binary (you may have to give its resolver some hints), and then you'd use that bootstrap `stack` binary to build the final `stack` binary.

**Comment by essandess** [comment]
Thanks. We also bootstrap `cabal` and `ghc`, and the `ghc` bootstrap needs `alex`, and `alex` needs … `stack`.

**Comment by shayne-fletcher** [comment]
I confirm that this candidate works nicely for ghc-lib. Sure would appreciate it if this could be released with some urgency - I can't afford the cost of building it from source in CI! Thank-you for for your fixes and hard work!

**Comment by essandess** [comment]
Yep, also deployed to bootstrap `stack` for the last 4 macOS releases, and the is binary working on plenty more before that: http://packages.macports.org/stack/

Thanks again!
